python/src/equistore/tensor.py

```diff
@@ -105,6 +105,72 @@ def __getitem__(self, *args) -> TensorBlock:
             )
         return self.block(*args)
 
+    def __eq__(self, other):
+        from equistore.operations import equal
+
+        return equal(self, other)
+
+    def __ne__(self, other):
+        from equistore.operations import equal
+
+        return not equal(self, other)
+
+    def __add__(self, other):
+        from equistore.operations import add
+
+        return add(self, other)
+
+    def __sub__(self, other):
+        from equistore.operations import subtract
+
+        return subtract(self, other)
+
+    def __mul__(self, other):
+        from equistore.operations import multiply
+
+        return multiply(self, other)
+
+    def __matmul__(self, other):
+        from equistore.operations import dot
+
+        return dot(self, other)
+
+    def __truediv__(self, other):
+        from equistore.operations import divide
+
+        return divide(self, other)
+
+    def __pow__(self, other):
+        from equistore.operations import pow
+
+        return pow(self, other)
+
+    def __iadd__(self, other):
```

Comment:
I would not do the in-place operations if they are not actually in-place

Reply:
Yes, I agree.